examples/worker.cpp

```diff
@@ -80,26 +80,24 @@ bool DefineFunctions(JSContext* cx, JS::Handle<JSObject*> global) {
   return true;
 }
 
-static void* WorkerMain(void* arg) {
-  JSRuntime* parentRuntime = reinterpret_cast<JSRuntime*>(arg);
-
+static void WorkerMain(JSRuntime* parentRuntime) {
   JSContext* cx = JS_NewContext(8L * 1024L * 1024L, parentRuntime);
 
   if (!JS::InitSelfHostedCode(cx)) {
-    return nullptr;
+    return;
```

Comment:
Should we do something else on failure of these functions instead of just returning? e.g. logging, aborting, or using `std::promise` to communicate success or failure to the main thread?

Reply:
I'll add logging.

communicating with main thread is beyond the scope of this example.
if we should have an example for communication, we'd better add separate example that communicates with main thread also for successful case.